### PR TITLE
Fixing hidden roles support in SecurityRolesType for 2.2 branch

### PR DESCRIPTION
**Pull request by nsgomez**
This is a backport for the 2.2 branch, based on the fix merged at @0c56664
into the master branch via pull request #522.

### CODE REVIEW

**Comment by nsgomez** [comment]
@kaiwa made a fork in September which patched this in 2.2, but hadn't made a PR. This just cherry-picks his changes.

**Comment by nsgomez** [comment]
ping @rande

**Comment by kaiwa** [comment]
:thumbsup:

**Comment by OskarStark** [comment]
ping @rande 

btw, is there no cs-fixer running? @rande @Soullivaneuh

File: Tests/Form/Transformer/RestoreRolesTransformerTest.php [hunk 1]
```diff
@@ -100,4 +100,48 @@ public function testReverseTransformAllowEmptyOriginalRoles()
...
+        $roleBuilder->expects($this->once())->method('getRoles')->will($this->returnValue(array($availableRoles, array())));
+        // user roles
+        $userRoles = array('ROLE_COMPANY_PERSONAL_MODERATOR', 'ROLE_COMPANY_NEWS_MODERATOR', 'ROLE_COMPANY_BOOKKEEPER');
+        $transformer = new RestoreRolesTransformer($roleBuilder);
+        $transformer->setOriginalRoles($userRoles);
+        // now we want to revoke role ROLE_COMPANY_PERSONAL_MODERATOR
+        $revokedRole = array_shift($userRoles);
+        $processedRoles = $transformer->reverseTransform($userRoles);
+        $this->assertNotContains($revokedRole, $processedRoles);
+    }
```
Comment (OskarStark) [inline Tests/Form/Transformer/RestoreRolesTransformerTest.php]: line break needed

**Comment by nsgomez** [comment]
Fixed line breaks @OskarStark

**Comment by OskarStark** [comment]
can you please squash your last to commits and keep the cherry picked one?

@rande i think we can then merge this PR

**Comment by nsgomez** [comment]
Done

**Comment by OskarStark** [comment]
Thank you @nsgomez

**Pull request merged**